cmpctircd/Configuration/OperatorElement.cs

```diff
@@ -1,70 +1,31 @@
 ﻿using System;
-using System.Configuration;
-using System.Xml;
 using System.Collections.Generic;
-using System.Linq;
 using System.ComponentModel;
+using System.Configuration;
 using System.Globalization;
+using System.Linq;
 
 namespace cmpctircd.Configuration {
     public class OperatorElement : ConfigurationElement {
```

Comment:
Something is deeply wrong with /oper, I have no idea why the configuration is effecting how these modes are getting set, but for some reason they are.

`/oper {user} {password}` now hits line 280 in QueryController.cs which causes an Out of Bounds exception (the if statement logic is wrong here which is why it gets the exception) - in master it does not hit line 280 when you `/oper` so it avoids the exception. SpacedArgs contains 2 arguments so it gets hit - in master, this ModeHandler in QueryController.cs never even gets ran when you do `/oper` so what on earth has gone on here?

After the exception, you don't get oper'd up and you join a null channel, setting some weird modes:
```
* Now talking on 
* Josh sets mode Jo on +o
* Josh sets mode Js on +o
* Josh sets mode Jh on +o
```
These modes are obviously the letters of my username, but I have no idea why this is happening.

If I manually do `/mode {user} +o` on both branches it hits line 280 correctly (with 3 args)

Reply:
### This PR's Config Object:

```
-		Config	{Microsoft.Extensions.Options.OptionsManager<cmpctircd.Configuration.Options.ConfigurationOptions>}	Microsoft.Extensions.Options.IOptions<cmpctircd.Configuration.Options.ConfigurationOptions> {Microsoft.Extensions.Options.OptionsManager<cmpctircd.Configuration.Options.ConfigurationOptions>}
-		Value	{cmpctircd.Configuration.Options.ConfigurationOptions}	cmpctircd.Configuration.Options.ConfigurationOptions
-		Advanced	{cmpctircd.Configuration.Options.AdvancedOptions}	cmpctircd.Configuration.Options.AdvancedOptions
-		Cloak	{cmpctircd.Configuration.Options.CloakOptions}	cmpctircd.Configuration.Options.CloakOptions
		DomainParts	3	int
		Full	false	bool
		Key	"cmpct"	string
		Prefix	"cmpct"	string
		MaxTargets	200	int
		PingTimeout	120	int
		RequirePongCookie	true	bool
		ResolveHostnames	true	bool
-		CModes	{cmpctircd.Configuration.ModeElement[2]}	cmpctircd.Configuration.ModeElement[]
-		[0]	{cmpctircd.Configuration.ModeElement}	cmpctircd.Configuration.ModeElement
		Name	"n"	string
		Param	""	string
-		[1]	{cmpctircd.Configuration.ModeElement}	cmpctircd.Configuration.ModeElement
		Name	"t"	string
		Param	""	string
		Description	"The C# IRC Server"	string
		Host	"irc.cmpct.info"	string
-		Loggers	{cmpctircd.Configuration.LoggerElement[2]}	cmpctircd.Configuration.LoggerElement[]
-		[0]	{cmpctircd.Configuration.LoggerElement}	cmpctircd.Configuration.LoggerElement
+		Attributes	Count = 0	System.Collections.Generic.Dictionary<string, string>
		Level	Debug	cmpctircd.LogType
		Type	Stdout	cmpctircd.LoggerType
-		[1]	{cmpctircd.Configuration.LoggerElement}	cmpctircd.Configuration.LoggerElement
+		Attributes	Count = 1	System.Collections.Generic.Dictionary<string, string>
		Level	Warn	cmpctircd.LogType
		Type	File	cmpctircd.LoggerType
		Network	"cmpct"	string
-		OperChan	{string[1]}	string[]
		[0]	""	string
-		Opers	{cmpctircd.Configuration.OperatorElement[1]}	cmpctircd.Configuration.OperatorElement[]
-		[0]	{cmpctircd.Configuration.OperatorElement}	cmpctircd.Configuration.OperatorElement
		Algorithm	"System.Security.Cryptography.SHA256Managed"	string
+		AlgorithmType	{Name = "SHA256Managed" FullName = "System.Security.Cryptography.SHA256Managed"}	System.Type {System.RuntimeType}
		CurrentConfiguration	null	System.Configuration.Configuration
		DataToWriteInternal	false	bool
+		ElementInformation	{System.Configuration.ElementInformation}	System.Configuration.ElementInformation
		ElementPresent	false	bool
+		ElementProperty	{System.Configuration.ConfigurationElementProperty}	System.Configuration.ConfigurationElementProperty
		ElementTagName	null	string
+		EvaluationContext	'((Microsoft.Extensions.Options.OptionsManager<cmpctircd.Configuration.Options.ConfigurationOptions>)Config).Value.Opers[0].EvaluationContext' threw an exception of type 'System.Configuration.ConfigurationErrorsException'	System.Configuration.ContextInformation {System.Configuration.ConfigurationErrorsException}
		HasContext	false	bool
+		Hosts	Count = 1	System.Collections.Generic.List<string>
		ItemLocked	Modified	System.Configuration.ConfigurationValueFlags
+		LockAllAttributesExcept	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockAllElementsExcept	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockAttributes	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockElements	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		LockItem	false	bool
+		LockedAllExceptAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockedAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		Name	"josh"	string
		Password	"5e884898da28047151d0e56f8dc6292773603d0d6aabbdd62a11ef721d1542d8"	string
+		PasswordArray	{byte[32]}	byte[]
+		Properties	{System.Configuration.ConfigurationPropertyCollection}	System.Configuration.ConfigurationPropertyCollection
		Tls	false	bool
+		Values	{System.Configuration.ConfigurationValues}	System.Configuration.ConfigurationValues
+		_configRecord	null	System.Configuration.BaseConfigurationRecord
+		_elementProperty	{System.Configuration.ConfigurationElementProperty}	System.Configuration.ConfigurationElementProperty
		_evalContext	null	System.Configuration.ContextInformation
+		_evaluationElement	{System.Configuration.ElementInformation}	System.Configuration.ElementInformation
		_initialized	false	bool
		_itemLockedFlag	Modified	System.Configuration.ConfigurationValueFlags
+		_lockedAllExceptAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedAllExceptElementsList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedElementsList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		_modified	false	bool
		_readOnly	false	bool
+		Static members		
-		Servers	{cmpctircd.Configuration.ServerElement[1]}	cmpctircd.Configuration.ServerElement[]
-		[0]	{cmpctircd.Configuration.ServerElement}	cmpctircd.Configuration.ServerElement
		Destination	null	string
		Host	"services.cmpct.info"	string
-		Masks	Count = 1	System.Collections.Generic.List<string>
		[0]	"*@127.0.0.1"	string
+		Raw View		
		Outbound	false	bool
		Password	"mypassword"	string
		Port	9000	int
		Tls	false	bool
		Type	InspIRCd20	cmpctircd.ServerType
		VerifyTlsCert	false	bool
		Sid	"auto"	string
-		Sockets	{cmpctircd.Configuration.SocketElement[1]}	cmpctircd.Configuration.SocketElement[]
-		[0]	{cmpctircd.Configuration.SocketElement}	cmpctircd.Configuration.SocketElement
-		EndPoint	{127.0.0.1:6667}	System.Net.IPEndPoint
+		Address	{127.0.0.1}	System.Net.IPAddress
		AddressFamily	InterNetwork	System.Net.Sockets.AddressFamily
		Port	6667	int
+		_address	{127.0.0.1}	System.Net.IPAddress
		_port	6667	int
+		Static members		
		Host	"127.0.0.1"	string
		Port	6667	int
		Protocol	Dummy	cmpctircd.ServerType
		Tls	false	bool
		Type	Client	cmpctircd.ListenerType
-		Tls	{cmpctircd.Configuration.Options.TlsOptions}	cmpctircd.Configuration.Options.TlsOptions
		File	"server.pfx"	string
		Password	""	string
-		UModes	{cmpctircd.Configuration.ModeElement[2]}	cmpctircd.Configuration.ModeElement[]
-		[0]	{cmpctircd.Configuration.ModeElement}	cmpctircd.Configuration.ModeElement
		Name	"x"	string
		Param	""	string
-		[1]	{cmpctircd.Configuration.ModeElement}	cmpctircd.Configuration.ModeElement
		Name	"i"	string
		Param	""	string
+		_cache	{Microsoft.Extensions.Options.OptionsCache<cmpctircd.Configuration.Options.ConfigurationOptions>}	Microsoft.Extensions.Options.OptionsCache<cmpctircd.Configuration.Options.ConfigurationOptions>
+		_factory	{Microsoft.Extensions.Options.OptionsFactory<cmpctircd.Configuration.Options.ConfigurationOptions>}	Microsoft.Extensions.Options.IOptionsFactory<cmpctircd.Configuration.Options.ConfigurationOptions> {Microsoft.Extensions.Options.OptionsFactory<cmpctircd.Configuration.Options.ConfigurationOptions>}
```

### Master Config:
```
-		Config	{cmpctircd.Configuration.CmpctConfigurationSection}	cmpctircd.Configuration.CmpctConfigurationSection
-		Advanced	{cmpctircd.Configuration.AdvancedElement}	cmpctircd.Configuration.AdvancedElement
-		Cloak	{cmpctircd.Configuration.CloakElement}	cmpctircd.Configuration.CloakElement
		CurrentConfiguration	null	System.Configuration.Configuration
		DataToWriteInternal	false	bool
		DomainParts	3	int
+		ElementInformation	{System.Configuration.ElementInformation}	System.Configuration.ElementInformation
		ElementPresent	true	bool
+		ElementProperty	{System.Configuration.ConfigurationElementProperty}	System.Configuration.ConfigurationElementProperty
		ElementTagName	"cloak"	string
+		EvaluationContext	{System.Configuration.ContextInformation}	System.Configuration.ContextInformation
		Full	false	bool
		HasContext	true	bool
		ItemLocked	Default	System.Configuration.ConfigurationValueFlags
		Key	"cmpct"	string
+		LockAllAttributesExcept	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockAllElementsExcept	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockAttributes	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockElements	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		LockItem	false	bool
+		LockedAllExceptAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockedAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		Prefix	"cmpct"	string
+		Properties	{System.Configuration.ConfigurationPropertyCollection}	System.Configuration.ConfigurationPropertyCollection
+		Values	{System.Configuration.ConfigurationValues}	System.Configuration.ConfigurationValues
+		_configRecord	ConfigPath = "MACHINE/EXE"	System.Configuration.BaseConfigurationRecord {System.Configuration.RuntimeConfigurationRecord}
+		_elementProperty	{System.Configuration.ConfigurationElementProperty}	System.Configuration.ConfigurationElementProperty
+		_evalContext	{System.Configuration.ContextInformation}	System.Configuration.ContextInformation
+		_evaluationElement	{System.Configuration.ElementInformation}	System.Configuration.ElementInformation
		_initialized	true	bool
		_itemLockedFlag	Default	System.Configuration.ConfigurationValueFlags
+		_lockedAllExceptAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedAllExceptElementsList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedElementsList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		_modified	false	bool
		_readOnly	true	bool
+		Static members		
		CurrentConfiguration	null	System.Configuration.Configuration
		DataToWriteInternal	false	bool
+		ElementInformation	{System.Configuration.ElementInformation}	System.Configuration.ElementInformation
		ElementPresent	true	bool
+		ElementProperty	{System.Configuration.ConfigurationElementProperty}	System.Configuration.ConfigurationElementProperty
		ElementTagName	"advanced"	string
+		EvaluationContext	{System.Configuration.ContextInformation}	System.Configuration.ContextInformation
		HasContext	true	bool
		ItemLocked	Default	System.Configuration.ConfigurationValueFlags
+		LockAllAttributesExcept	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockAllElementsExcept	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockAttributes	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockElements	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		LockItem	false	bool
+		LockedAllExceptAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockedAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		MaxTargets	200	int
		PingTimeout	120	int
+		Properties	{System.Configuration.ConfigurationPropertyCollection}	System.Configuration.ConfigurationPropertyCollection
		RequirePongCookie	true	bool
		ResolveHostnames	true	bool
+		Values	{System.Configuration.ConfigurationValues}	System.Configuration.ConfigurationValues
+		_configRecord	ConfigPath = "MACHINE/EXE"	System.Configuration.BaseConfigurationRecord {System.Configuration.RuntimeConfigurationRecord}
+		_elementProperty	{System.Configuration.ConfigurationElementProperty}	System.Configuration.ConfigurationElementProperty
+		_evalContext	{System.Configuration.ContextInformation}	System.Configuration.ContextInformation
+		_evaluationElement	{System.Configuration.ElementInformation}	System.Configuration.ElementInformation
		_initialized	true	bool
		_itemLockedFlag	Default	System.Configuration.ConfigurationValueFlags
+		_lockedAllExceptAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedAllExceptElementsList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedElementsList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		_modified	false	bool
		_readOnly	true	bool
+		Static members		
-		AutomaticModes	Count = 2	cmpctircd.Configuration.ModeElementCollection
		AddElementName	"mode"	string
		ClearElementName	"clear"	string
		CollectionType	AddRemoveClearMap	System.Configuration.ConfigurationElementCollectionType
		Count	2	int
		CurrentConfiguration	null	System.Configuration.Configuration
		DataToWriteInternal	false	bool
+		ElementInformation	{System.Configuration.ElementInformation}	System.Configuration.ElementInformation
		ElementName	""	string
		ElementPresent	true	bool
+		ElementProperty	{System.Configuration.ConfigurationElementProperty}	System.Configuration.ConfigurationElementProperty
		ElementTagName	"cmodes"	string
		EmitClear	false	bool
+		EvaluationContext	{System.Configuration.ContextInformation}	System.Configuration.ContextInformation
		HasContext	true	bool
		InternalAddToEnd	false	bool
		InternalElementTagName	""	string
		IsSynchronized	false	bool
		ItemLocked	Default	System.Configuration.ConfigurationValueFlags
+		Items	Count = 2	System.Collections.ArrayList
+		LockAllAttributesExcept	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockAllElementsExcept	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockAttributes	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockElements	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		LockItem	false	bool
		LockableElements	"'mode', 'remove', 'clear'"	string
+		LockedAllExceptAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockedAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		Properties	{System.Configuration.ConfigurationPropertyCollection}	System.Configuration.ConfigurationPropertyCollection
		RemoveElementName	"remove"	string
		SyncRoot	null	object
		ThrowOnDuplicate	true	bool
+		Values	{System.Configuration.ConfigurationValues}	System.Configuration.ConfigurationValues
		_addElement	"mode"	string
		_clearElement	"clear"	string
		_collectionCleared	false	bool
		_comparer	null	System.Collections.IComparer
+		_configRecord	ConfigPath = "MACHINE/EXE"	System.Configuration.BaseConfigurationRecord {System.Configuration.RuntimeConfigurationRecord}
+		_elementProperty	{System.Configuration.ConfigurationElementProperty}	System.Configuration.ConfigurationElementProperty
		_emitClearTag	false	bool
+		_evalContext	{System.Configuration.ContextInformation}	System.Configuration.ContextInformation
+		_evaluationElement	{System.Configuration.ElementInformation}	System.Configuration.ElementInformation
		_inheritedCount	0	int
		_initialized	true	bool
		_itemLockedFlag	Default	System.Configuration.ConfigurationValueFlags
+		_lockedAllExceptAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedAllExceptElementsList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedElementsList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		_modified (System.Configuration.ConfigurationElement)	false	bool
		_modified	false	bool
		_readOnly (System.Configuration.ConfigurationElement)	false	bool
		_readOnly	true	bool
		_removeElement	"remove"	string
		_removedItemCount	0	int
+		Static members		
+		Results View	Expanding the Results View will enumerate the IEnumerable	
-		AutomaticUserModes	Count = 2	cmpctircd.Configuration.ModeElementCollection
		AddElementName	"mode"	string
		ClearElementName	"clear"	string
		CollectionType	AddRemoveClearMap	System.Configuration.ConfigurationElementCollectionType
		Count	2	int
		CurrentConfiguration	null	System.Configuration.Configuration
		DataToWriteInternal	false	bool
+		ElementInformation	{System.Configuration.ElementInformation}	System.Configuration.ElementInformation
		ElementName	""	string
		ElementPresent	true	bool
+		ElementProperty	{System.Configuration.ConfigurationElementProperty}	System.Configuration.ConfigurationElementProperty
		ElementTagName	"umodes"	string
		EmitClear	false	bool
+		EvaluationContext	{System.Configuration.ContextInformation}	System.Configuration.ContextInformation
		HasContext	true	bool
		InternalAddToEnd	false	bool
		InternalElementTagName	""	string
		IsSynchronized	false	bool
		ItemLocked	Default	System.Configuration.ConfigurationValueFlags
+		Items	Count = 2	System.Collections.ArrayList
+		LockAllAttributesExcept	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockAllElementsExcept	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockAttributes	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockElements	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		LockItem	false	bool
		LockableElements	"'mode', 'remove', 'clear'"	string
+		LockedAllExceptAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockedAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		Properties	{System.Configuration.ConfigurationPropertyCollection}	System.Configuration.ConfigurationPropertyCollection
		RemoveElementName	"remove"	string
		SyncRoot	null	object
		ThrowOnDuplicate	true	bool
+		Values	{System.Configuration.ConfigurationValues}	System.Configuration.ConfigurationValues
		_addElement	"mode"	string
		_clearElement	"clear"	string
		_collectionCleared	false	bool
		_comparer	null	System.Collections.IComparer
+		_configRecord	ConfigPath = "MACHINE/EXE"	System.Configuration.BaseConfigurationRecord {System.Configuration.RuntimeConfigurationRecord}
+		_elementProperty	{System.Configuration.ConfigurationElementProperty}	System.Configuration.ConfigurationElementProperty
		_emitClearTag	false	bool
+		_evalContext	{System.Configuration.ContextInformation}	System.Configuration.ContextInformation
+		_evaluationElement	{System.Configuration.ElementInformation}	System.Configuration.ElementInformation
		_inheritedCount	0	int
		_initialized	true	bool
		_itemLockedFlag	Default	System.Configuration.ConfigurationValueFlags
+		_lockedAllExceptAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedAllExceptElementsList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedElementsList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		_modified (System.Configuration.ConfigurationElement)	false	bool
		_modified	false	bool
		_readOnly (System.Configuration.ConfigurationElement)	false	bool
		_readOnly	true	bool
		_removeElement	"remove"	string
		_removedItemCount	0	int
+		Static members		
+		Results View	Expanding the Results View will enumerate the IEnumerable	
		CurrentConfiguration	null	System.Configuration.Configuration
		DataToWriteInternal	false	bool
		Description	"The C# IRC Server"	string
+		ElementInformation	{System.Configuration.ElementInformation}	System.Configuration.ElementInformation
		ElementPresent	true	bool
+		ElementProperty	{System.Configuration.ConfigurationElementProperty}	System.Configuration.ConfigurationElementProperty
		ElementTagName	"ircd"	string
+		EvaluationContext	{System.Configuration.ContextInformation}	System.Configuration.ContextInformation
		HasContext	true	bool
		Host	"irc.cmpct.info"	string
		ItemLocked	Default	System.Configuration.ConfigurationValueFlags
+		LockAllAttributesExcept	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockAllElementsExcept	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockAttributes	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockElements	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		LockItem	false	bool
+		LockedAllExceptAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockedAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
-		Loggers	Count = 2	cmpctircd.Configuration.LoggerElementCollection
		AddElementName	"logger"	string
		ClearElementName	"clear"	string
		CollectionType	AddRemoveClearMap	System.Configuration.ConfigurationElementCollectionType
		Count	2	int
		CurrentConfiguration	null	System.Configuration.Configuration
		DataToWriteInternal	false	bool
+		ElementInformation	{System.Configuration.ElementInformation}	System.Configuration.ElementInformation
		ElementName	""	string
		ElementPresent	true	bool
+		ElementProperty	{System.Configuration.ConfigurationElementProperty}	System.Configuration.ConfigurationElementProperty
		ElementTagName	"log"	string
		EmitClear	false	bool
+		EvaluationContext	{System.Configuration.ContextInformation}	System.Configuration.ContextInformation
		HasContext	true	bool
		InternalAddToEnd	false	bool
		InternalElementTagName	""	string
		IsSynchronized	false	bool
		ItemLocked	Default	System.Configuration.ConfigurationValueFlags
+		Items	Count = 2	System.Collections.ArrayList
+		LockAllAttributesExcept	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockAllElementsExcept	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockAttributes	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockElements	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		LockItem	false	bool
		LockableElements	"'logger', 'remove', 'clear'"	string
+		LockedAllExceptAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockedAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		Properties	{System.Configuration.ConfigurationPropertyCollection}	System.Configuration.ConfigurationPropertyCollection
		RemoveElementName	"remove"	string
		SyncRoot	null	object
		ThrowOnDuplicate	true	bool
+		Values	{System.Configuration.ConfigurationValues}	System.Configuration.ConfigurationValues
		_addElement	"logger"	string
		_clearElement	"clear"	string
		_collectionCleared	false	bool
		_comparer	null	System.Collections.IComparer
+		_configRecord	ConfigPath = "MACHINE/EXE"	System.Configuration.BaseConfigurationRecord {System.Configuration.RuntimeConfigurationRecord}
+		_elementProperty	{System.Configuration.ConfigurationElementProperty}	System.Configuration.ConfigurationElementProperty
		_emitClearTag	false	bool
+		_evalContext	{System.Configuration.ContextInformation}	System.Configuration.ContextInformation
+		_evaluationElement	{System.Configuration.ElementInformation}	System.Configuration.ElementInformation
		_inheritedCount	0	int
		_initialized	true	bool
		_itemLockedFlag	Default	System.Configuration.ConfigurationValueFlags
+		_lockedAllExceptAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedAllExceptElementsList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedElementsList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		_modified (System.Configuration.ConfigurationElement)	false	bool
		_modified	false	bool
		_readOnly (System.Configuration.ConfigurationElement)	false	bool
		_readOnly	true	bool
		_removeElement	"remove"	string
		_removedItemCount	0	int
+		Static members		
+		Results View	Expanding the Results View will enumerate the IEnumerable	
		Network	"cmpct"	string
-		Operators	Count = 1	cmpctircd.Configuration.OperatorElementCollection
		AddElementName	"oper"	string
+		Channels	Count = 0	System.Collections.Generic.IList<string> {System.Collections.Generic.List<string>}
		ClearElementName	"clear"	string
		CollectionType	AddRemoveClearMap	System.Configuration.ConfigurationElementCollectionType
		Count	1	int
		CurrentConfiguration	null	System.Configuration.Configuration
		DataToWriteInternal	false	bool
+		ElementInformation	{System.Configuration.ElementInformation}	System.Configuration.ElementInformation
		ElementName	""	string
		ElementPresent	true	bool
+		ElementProperty	{System.Configuration.ConfigurationElementProperty}	System.Configuration.ConfigurationElementProperty
		ElementTagName	"opers"	string
		EmitClear	false	bool
+		EvaluationContext	{System.Configuration.ContextInformation}	System.Configuration.ContextInformation
		HasContext	true	bool
		InternalAddToEnd	false	bool
		InternalElementTagName	""	string
		IsSynchronized	false	bool
		ItemLocked	Default	System.Configuration.ConfigurationValueFlags
+		Items	Count = 1	System.Collections.ArrayList
+		LockAllAttributesExcept	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockAllElementsExcept	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockAttributes	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockElements	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		LockItem	false	bool
		LockableElements	"'oper', 'remove', 'clear'"	string
+		LockedAllExceptAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockedAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		Properties	{System.Configuration.ConfigurationPropertyCollection}	System.Configuration.ConfigurationPropertyCollection
		RemoveElementName	"remove"	string
		SyncRoot	null	object
		ThrowOnDuplicate	true	bool
+		Values	{System.Configuration.ConfigurationValues}	System.Configuration.ConfigurationValues
		_addElement	"oper"	string
		_clearElement	"clear"	string
		_collectionCleared	false	bool
		_comparer	null	System.Collections.IComparer
+		_configRecord	ConfigPath = "MACHINE/EXE"	System.Configuration.BaseConfigurationRecord {System.Configuration.RuntimeConfigurationRecord}
+		_elementProperty	{System.Configuration.ConfigurationElementProperty}	System.Configuration.ConfigurationElementProperty
		_emitClearTag	false	bool
+		_evalContext	{System.Configuration.ContextInformation}	System.Configuration.ContextInformation
+		_evaluationElement	{System.Configuration.ElementInformation}	System.Configuration.ElementInformation
		_inheritedCount	0	int
		_initialized	true	bool
		_itemLockedFlag	Default	System.Configuration.ConfigurationValueFlags
+		_lockedAllExceptAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedAllExceptElementsList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedElementsList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		_modified (System.Configuration.ConfigurationElement)	false	bool
		_modified	false	bool
		_readOnly (System.Configuration.ConfigurationElement)	false	bool
		_readOnly	true	bool
		_removeElement	"remove"	string
		_removedItemCount	0	int
+		Static members		
+		Results View	Expanding the Results View will enumerate the IEnumerable	
-		Properties	{System.Configuration.ConfigurationPropertyCollection}	System.Configuration.ConfigurationPropertyCollection
		Count	12	int
+		DefaultCollectionProperty	null	System.Configuration.ConfigurationProperty
		IsSynchronized	false	bool
+		SyncRoot	Count = 12	object {System.Collections.ArrayList}
+		_items	Count = 12	System.Collections.ArrayList
+		Results View	Expanding the Results View will enumerate the IEnumerable	
		SID	"auto"	string
-		SectionInformation	{System.Configuration.SectionInformation}	System.Configuration.SectionInformation
		AllowDefinition	Everywhere	System.Configuration.ConfigurationAllowDefinition
		AllowDefinitionModified	false	bool
		AllowExeDefinition	MachineToApplication	System.Configuration.ConfigurationAllowExeDefinition
		AllowExeDefinitionModified	false	bool
		AllowLocation	true	bool
		AllowLocationModified	false	bool
		AllowOverride	true	bool
		Attached	true	bool
		ConfigKey	"ircd"	string
		ConfigSource	""	string
		ConfigSourceModified	false	bool
		ConfigSourceStreamName	null	string
		ForceSave	false	bool
		InheritInChildApplications	true	bool
		IsDeclarationRequired	true	bool
		IsDeclared	true	bool
		IsDefinitionAllowed	true	bool
		IsLocked	false	bool
		IsProtected	false	bool
		IsRuntime	true	bool
		LocationAttributesAreDefault	true	bool
		Name	"ircd"	string
		OverrideMode	Inherit	System.Configuration.OverrideMode
		OverrideModeDefault	Allow	System.Configuration.OverrideMode
		OverrideModeDefaultModified	false	bool
+		OverrideModeDefaultSetting	{System.Configuration.OverrideModeSetting}	System.Configuration.OverrideModeSetting
		OverrideModeEffective	Allow	System.Configuration.OverrideMode
+		OverrideModeSetting	{System.Configuration.OverrideModeSetting}	System.Configuration.OverrideModeSetting
		ProtectionProvider	null	System.Configuration.ProtectedConfigurationProvider
		ProtectionProviderName	null	string
		RawXml	null	string
		Removed	false	bool
		RequirePermission	true	bool
		RequirePermissionModified	false	bool
		RestartOnExternalChanges	true	bool
		RestartOnExternalChangesModified	false	bool
		SectionName	"ircd"	string
		Type	"cmpctircd.Configuration.CmpctConfigurationSection,cmpctircd"	string
		_allowDefinition	Everywhere	System.Configuration.ConfigurationAllowDefinition
		_allowExeDefinition	MachineToApplication	System.Configuration.ConfigurationAllowExeDefinition
+		_configRecord	null	System.Configuration.MgmtConfigurationRecord
		_configSource	null	string
+		_configurationSection	{cmpctircd.Configuration.CmpctConfigurationSection}	System.Configuration.ConfigurationSection {cmpctircd.Configuration.CmpctConfigurationSection}
+		_flags	{System.Configuration.SafeBitVector32}	System.Configuration.SafeBitVector32
+		_modifiedFlags	{System.Configuration.SimpleBitVector32}	System.Configuration.SimpleBitVector32
+		_overrideMode	{System.Configuration.OverrideModeSetting}	System.Configuration.OverrideModeSetting
+		_overrideModeDefault	{System.Configuration.OverrideModeSetting}	System.Configuration.OverrideModeSetting
		_protectionProvider	null	System.Configuration.ProtectedConfigurationProvider
		_typeName	"cmpctircd.Configuration.CmpctConfigurationSection,cmpctircd"	string
+		Static members		
-		Servers	Count = 1	cmpctircd.Configuration.ServerElementCollection
		AddElementName	"server"	string
		ClearElementName	"clear"	string
		CollectionType	AddRemoveClearMap	System.Configuration.ConfigurationElementCollectionType
		Count	1	int
		CurrentConfiguration	null	System.Configuration.Configuration
		DataToWriteInternal	false	bool
+		ElementInformation	{System.Configuration.ElementInformation}	System.Configuration.ElementInformation
		ElementName	""	string
		ElementPresent	true	bool
+		ElementProperty	{System.Configuration.ConfigurationElementProperty}	System.Configuration.ConfigurationElementProperty
		ElementTagName	"servers"	string
		EmitClear	false	bool
+		EvaluationContext	{System.Configuration.ContextInformation}	System.Configuration.ContextInformation
		HasContext	true	bool
		InternalAddToEnd	false	bool
		InternalElementTagName	""	string
		IsSynchronized	false	bool
		ItemLocked	Default	System.Configuration.ConfigurationValueFlags
+		Items	Count = 1	System.Collections.ArrayList
+		LockAllAttributesExcept	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockAllElementsExcept	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockAttributes	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockElements	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		LockItem	false	bool
		LockableElements	"'server', 'remove', 'clear'"	string
+		LockedAllExceptAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockedAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		Properties	{System.Configuration.ConfigurationPropertyCollection}	System.Configuration.ConfigurationPropertyCollection
		RemoveElementName	"remove"	string
		SyncRoot	null	object
		ThrowOnDuplicate	true	bool
+		Values	{System.Configuration.ConfigurationValues}	System.Configuration.ConfigurationValues
		_addElement	"server"	string
		_clearElement	"clear"	string
		_collectionCleared	false	bool
		_comparer	null	System.Collections.IComparer
+		_configRecord	ConfigPath = "MACHINE/EXE"	System.Configuration.BaseConfigurationRecord {System.Configuration.RuntimeConfigurationRecord}
+		_elementProperty	{System.Configuration.ConfigurationElementProperty}	System.Configuration.ConfigurationElementProperty
		_emitClearTag	false	bool
+		_evalContext	{System.Configuration.ContextInformation}	System.Configuration.ContextInformation
+		_evaluationElement	{System.Configuration.ElementInformation}	System.Configuration.ElementInformation
		_inheritedCount	0	int
		_initialized	true	bool
		_itemLockedFlag	Default	System.Configuration.ConfigurationValueFlags
+		_lockedAllExceptAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedAllExceptElementsList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedElementsList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		_modified (System.Configuration.ConfigurationElement)	false	bool
		_modified	false	bool
		_readOnly (System.Configuration.ConfigurationElement)	false	bool
		_readOnly	true	bool
		_removeElement	"remove"	string
		_removedItemCount	0	int
+		Static members		
+		Results View	Expanding the Results View will enumerate the IEnumerable	
-		Sockets	Count = 1	cmpctircd.Configuration.SocketElementCollection
		AddElementName	"socket"	string
		ClearElementName	"clear"	string
		CollectionType	AddRemoveClearMap	System.Configuration.ConfigurationElementCollectionType
		Count	1	int
		CurrentConfiguration	null	System.Configuration.Configuration
		DataToWriteInternal	false	bool
+		ElementInformation	{System.Configuration.ElementInformation}	System.Configuration.ElementInformation
		ElementName	""	string
		ElementPresent	true	bool
+		ElementProperty	{System.Configuration.ConfigurationElementProperty}	System.Configuration.ConfigurationElementProperty
		ElementTagName	"sockets"	string
		EmitClear	false	bool
+		EvaluationContext	{System.Configuration.ContextInformation}	System.Configuration.ContextInformation
		HasContext	true	bool
		InternalAddToEnd	false	bool
		InternalElementTagName	""	string
		IsSynchronized	false	bool
		ItemLocked	Default	System.Configuration.ConfigurationValueFlags
+		Items	Count = 1	System.Collections.ArrayList
+		LockAllAttributesExcept	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockAllElementsExcept	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockAttributes	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockElements	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		LockItem	false	bool
		LockableElements	"'socket', 'remove', 'clear'"	string
+		LockedAllExceptAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockedAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		Properties	{System.Configuration.ConfigurationPropertyCollection}	System.Configuration.ConfigurationPropertyCollection
		RemoveElementName	"remove"	string
		SyncRoot	null	object
		ThrowOnDuplicate	true	bool
+		Values	{System.Configuration.ConfigurationValues}	System.Configuration.ConfigurationValues
		_addElement	"socket"	string
		_clearElement	"clear"	string
		_collectionCleared	false	bool
		_comparer	null	System.Collections.IComparer
+		_configRecord	ConfigPath = "MACHINE/EXE"	System.Configuration.BaseConfigurationRecord {System.Configuration.RuntimeConfigurationRecord}
+		_elementProperty	{System.Configuration.ConfigurationElementProperty}	System.Configuration.ConfigurationElementProperty
		_emitClearTag	false	bool
+		_evalContext	{System.Configuration.ContextInformation}	System.Configuration.ContextInformation
+		_evaluationElement	{System.Configuration.ElementInformation}	System.Configuration.ElementInformation
		_inheritedCount	0	int
		_initialized	true	bool
		_itemLockedFlag	Default	System.Configuration.ConfigurationValueFlags
+		_lockedAllExceptAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedAllExceptElementsList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedElementsList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		_modified (System.Configuration.ConfigurationElement)	false	bool
		_modified	false	bool
		_readOnly (System.Configuration.ConfigurationElement)	false	bool
		_readOnly	true	bool
		_removeElement	"remove"	string
		_removedItemCount	0	int
+		Static members		
+		Results View	Expanding the Results View will enumerate the IEnumerable	
-		Tls	{cmpctircd.Configuration.TlsElement}	cmpctircd.Configuration.TlsElement
		CurrentConfiguration	null	System.Configuration.Configuration
		DataToWriteInternal	false	bool
+		ElementInformation	{System.Configuration.ElementInformation}	System.Configuration.ElementInformation
		ElementPresent	true	bool
+		ElementProperty	{System.Configuration.ConfigurationElementProperty}	System.Configuration.ConfigurationElementProperty
		ElementTagName	"tls"	string
+		EvaluationContext	{System.Configuration.ContextInformation}	System.Configuration.ContextInformation
		File	"server.pfx"	string
		HasContext	true	bool
		ItemLocked	Default	System.Configuration.ConfigurationValueFlags
+		LockAllAttributesExcept	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockAllElementsExcept	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockAttributes	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockElements	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		LockItem	false	bool
+		LockedAllExceptAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		LockedAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		Password	""	string
+		Properties	{System.Configuration.ConfigurationPropertyCollection}	System.Configuration.ConfigurationPropertyCollection
+		Values	{System.Configuration.ConfigurationValues}	System.Configuration.ConfigurationValues
+		_configRecord	ConfigPath = "MACHINE/EXE"	System.Configuration.BaseConfigurationRecord {System.Configuration.RuntimeConfigurationRecord}
+		_elementProperty	{System.Configuration.ConfigurationElementProperty}	System.Configuration.ConfigurationElementProperty
+		_evalContext	{System.Configuration.ContextInformation}	System.Configuration.ContextInformation
+		_evaluationElement	{System.Configuration.ElementInformation}	System.Configuration.ElementInformation
		_initialized	true	bool
		_itemLockedFlag	Default	System.Configuration.ConfigurationValueFlags
+		_lockedAllExceptAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedAllExceptElementsList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedElementsList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		_modified	false	bool
		_readOnly	true	bool
+		Static members		
+		Values	{System.Configuration.ConfigurationValues}	System.Configuration.ConfigurationValues
+		_configRecord	ConfigPath = "MACHINE/EXE"	System.Configuration.BaseConfigurationRecord {System.Configuration.RuntimeConfigurationRecord}
+		_elementProperty	{System.Configuration.ConfigurationElementProperty}	System.Configuration.ConfigurationElementProperty
+		_evalContext	{System.Configuration.ContextInformation}	System.Configuration.ContextInformation
+		_evaluationElement	{System.Configuration.ElementInformation}	System.Configuration.ElementInformation
		_initialized	true	bool
		_itemLockedFlag	Default	System.Configuration.ConfigurationValueFlags
+		_lockedAllExceptAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedAllExceptElementsList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedAttributesList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
+		_lockedElementsList	{System.Configuration.ConfigurationLockCollection}	System.Configuration.ConfigurationLockCollection
		_modified	false	bool
		_readOnly	true	bool
+		Static members		

```

Reply:
It looks like the issue was the fact that my configuration didn't have a default operChan set, it was just an empty string. My validation missed this because the collection wasn't empty, but the values inside it were. The validation now validates both the collection and the inner values, and a default channel is included in the configuration.

After doing this, the behaviour of `/oper` is not consistent across both branches.